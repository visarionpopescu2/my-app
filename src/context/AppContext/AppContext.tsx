import React from 'react';
import { APP_STATE } from "types";

const initialState: APP_STATE = {
  user: undefined,
  articles: [],
  setArticles: () => {}
}

const AppContext = React.createContext(initialState);

export default AppContext
