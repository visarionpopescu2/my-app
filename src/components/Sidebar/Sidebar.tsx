import React, { useContext } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { Menu, Avatar, Space, Typography } from 'antd';
import { ROUTES } from 'routes';
import AppContext from 'context/AppContext/AppContext';


const avatarStyle: React.CSSProperties = {
  display: 'flex',
  justifyContent: 'center',
  flexDirection: 'column',
  padding: '40px 0',
};
const menuStyle: React.CSSProperties = {
  backgroundColor: '#3ba0e9',
  color: '#fff',
  padding: '20px 0',
};

function Sidebar() {
  const { user } = useContext(AppContext);
  let navigate = useNavigate();
  const selectedPage = useLocation().pathname;
  
  const MENU_ITEMS = [
    { key: '1', label: 'Dashboard', onClick: () => navigate(ROUTES.Dashboard) },
    { key: '2', label: 'Articles', onClick: () => navigate(ROUTES.Articles) },
  ];
  
  const highlight = () => {
    if (selectedPage === ROUTES.Dashboard) {
      return ['1'];
    } else if (selectedPage.includes(ROUTES.Articles)) {
      return ['2'];
    }
  }
  
  return (
    <div>
      <Space style={avatarStyle}>
        <Avatar size={64} src={process.env.REACT_APP_USER_IMG_URL} style={{}} />
        <Typography.Text style={{color: '#fff'}}>Hello,</Typography.Text>
        <Typography.Text strong style={{color: '#fff'}}>{user?.name}</Typography.Text>
      </Space>

      <Menu
        defaultSelectedKeys={['1']}
        selectedKeys={highlight()}
        mode="inline"
        style={menuStyle}
        items={MENU_ITEMS}
      />
    </div>
    
  );
};

export default Sidebar;
