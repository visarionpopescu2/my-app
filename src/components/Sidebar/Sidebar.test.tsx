import { render } from '@testing-library/react';
import Sidebar from './Sidebar';
import { MemoryRouter } from 'react-router-dom';
import AppContext from 'context/AppContext/AppContext';

const USER = {
  id: '1',
  name: 'name',
  username: 'username',
  email: 'email',
  phone: '123445',
  website: 'www.website.com',
  company: { name: 'company', bs: 'business', catchPhrase: 'description' },
  address: { street: 'street', suite: 'suite', zipcode: '1234', city: 'city', geo: { lat: '11', lng: '22' }}
};
const setArticles = jest.fn();

describe('Sidebar', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should render Sidebar component', () => {
    let view = render(
      <MemoryRouter>
        <AppContext.Provider value={{ user: USER, articles: [], setArticles: setArticles }}>
          <Sidebar />
        </AppContext.Provider>
      </MemoryRouter>
    )
    expect(view.asFragment()).toMatchSnapshot();
  });
});
