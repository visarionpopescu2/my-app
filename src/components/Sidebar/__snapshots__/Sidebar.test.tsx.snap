// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render Sidebar component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ant-space css-dev-only-do-not-override-2i2tap ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      style="display: flex; justify-content: center; flex-direction: column; padding: 40px 0px;"
    >
      <div
        class="ant-space-item"
      >
        <span
          class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-2i2tap"
          style="width: 64px; height: 64px; line-height: 64px; font-size: 18px;"
        >
          <img
            src="https://shorturl.at/dzBW2"
          />
        </span>
      </div>
      <div
        class="ant-space-item"
      >
        <span
          class="ant-typography css-dev-only-do-not-override-2i2tap"
          style="color: rgb(255, 255, 255);"
        >
          Hello,
        </span>
      </div>
      <div
        class="ant-space-item"
      >
        <span
          class="ant-typography css-dev-only-do-not-override-2i2tap"
          style="color: rgb(255, 255, 255);"
        >
          <strong>
            name
          </strong>
        </span>
      </div>
    </div>
    <ul
      class="ant-menu ant-menu-root ant-menu-inline ant-menu-light css-dev-only-do-not-override-2i2tap"
      data-menu-list="true"
      role="menu"
      style="background-color: rgb(59, 160, 233); color: rgb(255, 255, 255); padding: 20px 0px;"
      tabindex="0"
    >
      <li
        class="ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
        data-menu-id="rc-menu-uuid-test-1"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          Dashboard
        </span>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        data-menu-id="rc-menu-uuid-test-2"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          Articles
        </span>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>
</DocumentFragment>
`;
